mangopy/mango.py

```diff
@@ -0,0 +1,187 @@
+# mango.py
+# Basic data and plotting utilities for MANGO network
+#
+# created 2019-3-19 by LLamarche
+# Notes:
+# - Default data directory is TEMP/MANGOData/, where TEMP is defined by tempfile.gettempdir() (https://docs.python.org/3/library/tempfile.html)
+# - TODO: Data files availabe at ftp://isr.sri.com/pub/earthcube/provider/asti/MANGOProcessed/
+
+import numpy as np
+import datetime as dt
+import h5py
+import csv
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import os
+import urllib
+from contextlib import closing
+import tempfile
+
+
+class Mango(object):
+
+    def __init__(self, datadir=None):
+
+        self.mangopy_path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+        # if no data directory specified, use a default temp directory
+        if datadir is None:
+            datadir = os.path.join(tempfile.gettempdir(),'MANGOData')
+        self.datadir = datadir
+
+
+    def plot(self,site,targtime):
+        # plot single mango image
+        img, __, __, truetime = self.get_data(site,targtime)
+        plt.imshow(img, cmap=plt.get_cmap('gist_heat'))
+        plt.title('{:%Y-%m-%d %H:%M}'.format(truetime))
+        plt.show()
+
+    def map(self,site,targtime):
+        # map single mango image
+        img, lat, lon, truetime = self.get_data(site,targtime)
+
+        # set up map
+        fig = plt.figure()
+        map_proj = ccrs.LambertConformal(central_longitude=np.nanmean(lon),central_latitude=np.nanmean(lat))
+        ax = fig.add_subplot(111,projection=map_proj)
+        ax.coastlines()
+        ax.gridlines()
+        ax.add_feature(cfeature.STATES)
+        ax.set_extent([np.nanmin(lon),np.nanmax(lon),np.nanmin(lat),np.nanmax(lat)])
+
+        # plot image on map
+        ax.scatter(lon, lat, c=img, s=0.7, cmap=plt.get_cmap('gist_heat'), transform=ccrs.Geodetic())
+
+        # add time as title of plot
+        ax.set_title('{:%Y-%m-%d %H:%M}'.format(truetime))
+
+        plt.show()
+
+    def get_data(self,site,targtime):
+        # read mango data file
+        filename = os.path.join(self.datadir,'{0}/{1:%b%d%y}/{2}{1:%b%d%y}.h5'.format(site['name'],targtime,site['code']))
+
+        # first try to read data file locally
+        try:
+            img_array, lat, lon, truetime = self.read_datafile(filename,targtime)
+        # if that fails, try to download, then read the data file
+        except OSError:
```

Comment:
Perhaps one should be verbose about this? Print something out saying that the file needs to be fetched?

Reply:
Sounds good!